sdv/_utils.py

```diff
@@ -200,3 +201,80 @@ def _format_invalid_values_string(invalid_values, num_values):
             return f'{invalid_values[:num_values] + extra_missing_values}'
 
     return f'{invalid_values}'
+
+
+def _find_root_tables(relationships):
+    parent_tables = {rel['parent_table_name'] for rel in relationships}
+    child_tables = {rel['child_table_name'] for rel in relationships}
+    return parent_tables - child_tables
+
+
+def _get_relationship_idx_for_child(relationships, child_table):
+    return [idx for idx, rel in enumerate(relationships) if rel['child_table_name'] == child_table]
+
+
+def _get_relationship_idx_for_parent(relationships, parent_table):
+    return [
+        idx for idx, rel in enumerate(relationships) if rel['parent_table_name'] == parent_table
+    ]
+
+
+def _get_rows_to_drop(metadata, data):
```

Comment:
Should keep these as hidden functions in `utils.py` since they're helpers for `drop_unknown_references` and not generic utility functions?

Reply:
I think it's okay like this, some methods may be reused in the future.